### PR TITLE
python3.pkgs.geoarrow-pandas: init at 0.1.2

### PR DESCRIPTION
**Pull request by cpcloud**
New packages:

- **python3.pkgs.geoarrow-types: init at 0.2.0**
- **python3.pkgs.geoarrow-c: init at 0.1.3**
- **python3.pkgs.geoarrow-pyarrow: init at 0.1.2**
- **python3.pkgs.geoarrow-pandas: init at 0.1.2**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Sigmanificient** [comment]
Hi, can you take a look at this pr? I tried to package `quackosm` which depend on some `geoarrow` libs, but i could get to build the rust lib:

- https://github.com/NixOS/nixpkgs/pull/328819

**Comment by Sigmanificient** [comment]
Hi, can you take a look at this pr? I tried to package `quackosm` which depend on some `geoarrow` libs, but i could get to build the rust lib:

- https://github.com/NixOS/nixpkgs/pull/328819

File: pkgs/development/python-modules/geoarrow-c/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  nativeCheckInputs = [
+    pytestCheckHook
+    pyarrow
+    numpy
+  ];
+
+  pythonImportsCheck = [ "geoarrow.c" ];
+
+  meta = with lib; {
+    description = "Geoarrow types";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/geoarrow-c/default.nix:46]:
```suggestion
    description = "Experimental C and C++ implementation of the GeoArrow specification";
```

File: pkgs/development/python-modules/geoarrow-pandas/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    hash = "sha256-Ni+GKTRhRDRHip1us3OZPuUhHQCNU7Nap865T/+CU8Y=";
+  };
+
+  sourceRoot = "${src.name}/geoarrow-pandas";
+
+  nativeBuildInputs = [ setuptools-scm ];
+
+  env.SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  propagatedBuildInputs = [
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/geoarrow-pandas/default.nix:32]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/geoarrow-pandas/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "geoarrow.pandas" ];
+
+  meta = with lib; {
+    description = "Geoarrow pandas";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/geoarrow-pandas/default.nix:45]:
```suggestion
    description = "Python implementation of the GeoArrow specification";
```

File: pkgs/development/python-modules/geoarrow-pyarrow/default.nix [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  src = fetchFromGitHub {
+    repo = "geoarrow-python";
+    owner = "geoarrow";
+    rev = "refs/tags/geoarrow-pyarrow-${version}";
+    hash = "sha256-Ni+GKTRhRDRHip1us3OZPuUhHQCNU7Nap865T/+CU8Y=";
+  };
+
+  sourceRoot = "${src.name}/geoarrow-pyarrow";
+
+  nativeBuildInputs = [ setuptools-scm ];
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/geoarrow-pyarrow/default.nix:33]:
```suggestion
  build-system = [ setuptools-scm ];
```

File: pkgs/development/python-modules/geoarrow-pyarrow/default.nix [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    "test_multipoint_with_offset"
+    "test_multilinestring_with_offset"
+    "test_multipolygon_with_offset"
+    "test_multipolygon_with_offset_nonempty_inner_lists"
+    "test_interleaved_multipolygon_with_offset"
+    "test_readpyogrio_table_gpkg"
+    "test_geometry_type_basic"
+  ];
+
+  propagatedBuildInputs = [
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/geoarrow-pyarrow/default.nix:50]:
```suggestion
  dependencies = [
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Comment by cpcloud** [comment]
@fabaff @mweinelt Can one of y'all give this a review?

File: pkgs/development/python-modules/geoarrow-pandas/default.nix [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  src = fetchFromGitHub {
+    repo = "geoarrow-python";
+    owner = "geoarrow";
+    rev = "refs/tags/geoarrow-pandas-${version}";
+    hash = "sha256-Ni+GKTRhRDRHip1us3OZPuUhHQCNU7Nap865T/+CU8Y=";
+  };
+
+  sourceRoot = "${src.name}/geoarrow-pandas";
+
+  nativeBuildInputs = [ setuptools-scm ];
```
Comment (fabaff) [inline pkgs/development/python-modules/geoarrow-pandas/default.nix:28]:
```suggestion
  build-system = [ setuptools-scm ];
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 349312` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python311Packages.geoarrow-c</li>
    <li>python311Packages.geoarrow-c.dist</li>
    <li>python311Packages.geoarrow-pandas</li>
    <li>python311Packages.geoarrow-pandas.dist</li>
    <li>python311Packages.geoarrow-pyarrow</li>
    <li>python311Packages.geoarrow-pyarrow.dist</li>
    <li>python311Packages.geoarrow-types</li>
    <li>python311Packages.geoarrow-types.dist</li>
    <li>python312Packages.geoarrow-c</li>
    <li>python312Packages.geoarrow-c.dist</li>
    <li>python312Packages.geoarrow-pandas</li>
    <li>python312Packages.geoarrow-pandas.dist</li>
    <li>python312Packages.geoarrow-pyarrow</li>
    <li>python312Packages.geoarrow-pyarrow.dist</li>
    <li>python312Packages.geoarrow-types</li>
    <li>python312Packages.geoarrow-types.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/geoarrow-c/default.nix [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  src = fetchFromGitHub {
+    repo = "geoarrow-c";
+    owner = "geoarrow";
+    rev = "refs/tags/geoarrow-c-python-${version}";
+    hash = "sha256-kQCD3Vptl7GtRFigr4darvdtwnaHRLZWvBBpZ0xHMgM=";
+  };
+
+  sourceRoot = "${src.name}/python/geoarrow-c";
+
+  nativeBuildInputs = [
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply pkgs/development/python-modules/geoarrow-c/default.nix]: Done!

File: pkgs/development/python-modules/geoarrow-pandas/default.nix (hunk 6 above)
Comment (cpcloud) [reply pkgs/development/python-modules/geoarrow-pandas/default.nix]: Done!

**Comment by cpcloud** [comment]
@fabaff can this be merged?

**Comment by FliegendeWurst** [comment]
@ofborg build python312Packages.geoarrow-types

(timeout)

**Review by FliegendeWurst — APPROVED** [review]

**Comment by fabaff** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349312`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.geoarrow-c</li>
    <li>python311Packages.geoarrow-c.dist</li>
    <li>python311Packages.geoarrow-pandas</li>
    <li>python311Packages.geoarrow-pandas.dist</li>
    <li>python311Packages.geoarrow-pyarrow</li>
    <li>python311Packages.geoarrow-pyarrow.dist</li>
    <li>python311Packages.geoarrow-types</li>
    <li>python311Packages.geoarrow-types.dist</li>
    <li>python312Packages.geoarrow-c</li>
    <li>python312Packages.geoarrow-c.dist</li>
    <li>python312Packages.geoarrow-pandas</li>
    <li>python312Packages.geoarrow-pandas.dist</li>
    <li>python312Packages.geoarrow-pyarrow</li>
    <li>python312Packages.geoarrow-pyarrow.dist</li>
    <li>python312Packages.geoarrow-types</li>
    <li>python312Packages.geoarrow-types.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Pull request merged**